python/instrumentation/openinference-instrumentation-langchain/src/openinference/instrumentation/langchain/__init__.py

```diff
@@ -104,3 +118,20 @@ def get_current_span() -> Optional[Span]:
     if not run_id:
         return None
     return LangChainInstrumentor().get_span(run_id)
+
+
+def get_current_root_chain_span() -> Optional[Span]:
+    import langchain_core
+
+    run_id: Optional[UUID] = None
+    config = langchain_core.runnables.config.var_child_runnable_config.get()
+    if not isinstance(config, dict):
+        return None
+    for v in config.values():
+        if not isinstance(v, langchain_core.callbacks.BaseCallbackManager):
+            continue
+        if run_id := v.parent_run_id:
+            break
+    if not run_id:
+        return None
```

Comment:
If you have time, would you mind DRY-ing this up, since it's the same as line 107-119. Thx!

Reply:
I'm just pointing out that this is a simple copy-paste. I'm not suggesting adding abstractions or anything like that. But it's up to you.

Reply:
> my preference for first implementations

I’m suggesting that if you don’t anticipate making further updates to this repo in the near future, it’s generally a good practice to replace the copy-pasted sections with a helper function. However, if you think it’s better to leave that for the next maintainer, that’s also fine.